### DISCUSSION TITLE
Buttons should support additional form* attributes

### DISCUSSION THREAD

**Discussion by austinw-fineart** [material-components/material-web#5593] [Ideas]
### Description

Add support for [formaction](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button#formaction), [formenctype](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button#formenctype), [formmethod](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button#formmethod), [formnovalidate](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button#formnovalidate), [formtarget](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button#formtarget). These are used for button differentiation inside native forms. While you're at it, there's also the newer attributes [popovertarget](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button#popovertarget) and [popovertargetaction](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button#popovertargetaction).

**Comment by asyncliz** [comment]
The popover attributes are natively added to all HTML elements. You should be able to use `<md-filled-button popovertarget="id-ref">` on browsers that support it today.

Those extra form attributes are used to control aspects of the `<form>` itself. The name and value attributes on a button are used to differentiate which button submitted the form.

Can you give more use cases for when you need these attributes on the buttons themselves? The `<form>`'s attributes (`action`, `method`, `novalidate`) should generally provide this functionality. Having individual button submissions change the behavior of a form can be a brittle implementation. Performing different logic in the form's submission itself is typically more recommended.

**Reply by austinw-fineart** [reply]
I cannot seem to get the popovertarget attribute working?
[Example Playground](https://lit.dev/playground/#project=W3sibmFtZSI6Im1haW4udHMiLCJjb250ZW50IjoiaW1wb3J0ICdAbWF0ZXJpYWwvd2ViL2J1dHRvbi9maWxsZWQtYnV0dG9uLmpzJztcbmltcG9ydCB7aHRtbCwgY3NzLCBMaXRFbGVtZW50fSBmcm9tICdsaXQnO1xuaW1wb3J0IHtjdXN0b21FbGVtZW50LCBxdWVyeX0gZnJvbSAnbGl0L2RlY29yYXRvcnMuanMnO1xuXG5AY3VzdG9tRWxlbWVudCgnbXktY29tcG9uZW50JylcbmV4cG9ydCBjbGFzcyBNeUNvbXBvbmVudCBleHRlbmRzIExpdEVsZW1lbnQge1xuICBzdGF0aWMgc3R5bGVzID0gY3NzYGA7XG5cbiAgcmVuZGVyKCkge1xuICAgIHJldHVybiBodG1sYFxuICAgICAgPHNwYW4gc3BhbiBzdHlsZT1cInBvc2l0aW9uOiByZWxhdGl2ZVwiPlxuICAgICAgICA8YnV0dG9uXG4gICAgICAgICAgcG9wb3ZlcnRhcmdldD1cIm15cG9wb3ZlclwiPlxuICAgICAgICAgIEJ1dHRvblxuICAgICAgICA8L2J1dHRvbj5cbiAgICAgICAgPG1kLWZpbGxlZC1idXR0b25cbiAgICAgICAgICBwb3BvdmVydGFyZ2V0PVwibXlwb3BvdmVyXCI-XG4gICAgICAgICAgQnV0dG9uXG4gICAgICAgIDwvbWQtZmlsbGVkLWJ1dHRvbj5cbiAgICAgIDwvc3Bhbj5cbiAgICAgIDxkaXYgaWQ9XCJteXBvcG92ZXJcIiBwb3BvdmVyPlBvcG92ZXIgY29udGVudDwvZGl2PlxuICAgICAgICBgO1xuICB9XG59In0seyJuYW1lIjoiaW5kZXguaHRtbCIsImNvbnRlbnQiOiI8IURPQ1RZUEUgaHRtbD5cbjxoZWFkPlxuICA8c2NyaXB0IHR5cGU9XCJtb2R1bGVcIiBzcmM9XCIuL21haW4uanNcIj48L3NjcmlwdD5cbjwvaGVhZD5cbjxzdHlsZT5cbiAgXG48L3N0eWxlPlxuPGJvZHk-XG4gIDxteS1jb21wb25lbnQ-PC9teS1jb21wb25lbnQ-XG48L2JvZHk-XG4ifSx7Im5hbWUiOiJwYWNrYWdlLmpzb24iLCJjb250ZW50Ijoie1xuICBcImRlcGVuZGVuY2llc1wiOiB7XG4gICAgXCJsaXRcIjogXCJeMi4wLjBcIixcbiAgICBcIkBsaXQvcmVhY3RpdmUtZWxlbWVudFwiOiBcIl4xLjAuMFwiLFxuICAgIFwibGl0LWVsZW1lbnRcIjogXCJeMy4wLjBcIixcbiAgICBcImxpdC1odG1sXCI6IFwiXjIuMC4wXCJcbiAgfVxufSIsImhpZGRlbiI6dHJ1ZX1d)

As for the form attributes, they've long existed for the purpose of serving good ole server-driven forms. Having multiple buttons in a form would've served for interactivity in a time before interactive UIs were a thing. You could've written up a form in as little as:
```html
<form method="post">
  <md-outlined-text-field label="Hello" required></md-outlined-text-field>
  <md-filled-button formaction="/Search" formmethod="get">Search</md-filled-button>
  <md-filled-button formaction="/World" novalidate>World</md-filled-button>
  <md-filled-button formaction="/New">New</md-filled-button>
</form>
```
Yeah it's brittle, but that's what server validation is for. Are there any real world use cases? Dunno, ASP.NET certainly uses them; it's how I discovered the attributes. I just assumed these could be drop-in replacements for regular buttons so it'd be odd if this gap in support existed.

**Comment by jlesquembre** [comment]
Related: https://github.com/openui/open-ui/issues/1088
